packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -50,22 +51,26 @@ export async function monitorTransactionsProposedOrderBook(
   const markets = await getPolymarketMarkets(params);
 
   const marketsWithAncillary = await getMarketsAncillary(params, markets);
+
   // Filter out markets that do not have a proposal event.
   const marketsWithEventData: PolymarketWithEventData[] = marketsWithAncillary
-    .filter((market) => proposalEvents.find((event) => event.ancillaryData === market.ancillaryData))
+    .filter((market) =>
+      proposalEvents.find(
+        (event) =>
+          [params.binaryAdapterAddress, params.ctfAdapterAddress].includes(event.requester) &&
```

Comment:
Sorry to be annoying, but are we not able to get the _precise address_ that was used in this case? Just trying to make this 100% bulletproof in the face of spoofs or duplicates. If not, that's totally okay, just wanted to check.

Reply:
@mrice32 I'm now [filtering by market.resolvedBy ](https://github.com/UMAprotocol/protocol/blob/master/packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts#L61)

Also I've added a new check, if the polymarket api says a market is "proposed" but we don't find the event we notify always. This shouldn't happen but I thought it could be safer to have it if the api for some reason returns inconsistent information.

Reply:
+1!